Assets/Scripts/PlayerScripts/PlayerHealth.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PlayerHealth : Health
+{
+    private PlayerManager player;
+
+    void Setup()
+    {
+        this.currentHealth = this.maxHealth;
+        player = this.GetComponent<PlayerManager>();
+        player.onPlayerDeath += DestroyPlayer;
+    }
+
+    private void Awake()
+    {
+        Setup();
+    }
+
+    public override void DestroyOnDeath()
+    {
+        if(this.currentHealth <= 0)
+        {
+            player.HandlePlayerDeath();
+        }
+    }
+
+    public void HandleAbilityHit(AbilityBase ability)

```

Comment:
Still need to hook this up somehow. Works like before presently but it's not connected to the eve nt system like it should be. If `PlayerHealth` is the spot to do it that's fine. but it does feel a bit awkward

Reply:
If so, would simply access a public starter method on the base `Health` and then override in Player to call the `PlayerManager.HandlePlayerHitByAbility` function. In `Setup()` here, we would simply subscribe `HandleAbilityHit` to the delegate `onPlayerHitByAbility` in `PlayerManager`.

Essentially, those variables of the type delegate are like lists of functions. You can add a function to the list of them to be called. So when `Handle____` is eventually called you can have multiple subsystems handle different aspects of an event with a centralized system. 

eg: we might have some UI that does something when a player dies. or the arena might want to know about it or something. Those can simply subscribe to the player death delegate or we can have another GameManager that knows about all the Player Managers.

It is admittedly a little over-architect-ed for where we are now, but it is a really solid base + really solid design pattern for this. Esp. for multiplayer since each player can have there own even system with no additional effort from us!